### PR TITLE
Ensure SyncRoot is not null

### PR DESCRIPTION
**Pull request by twsouthwick**

### CODE REVIEW

File: src/Microsoft.AspNetCore.SystemWebAdapters/Internal/NonGenericCollectionWrapper.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public NonGenericCollectionWrapper(ICollection<T> collection)
 
         public bool IsSynchronized => false;
 
-        public object SyncRoot => null!;
+        public object SyncRoot => _collection;
```
Comment (mjrousos) [inline src/Microsoft.AspNetCore.SystemWebAdapters/Internal/NonGenericCollectionWrapper.cs:23]: I think this is very unlikely to matter one way or another, but I looked at how we've done this in the framework before and there appears to be some precedent for just creating a new syncroot object if the wrapped collection doesn't have one: https://referencesource.microsoft.com/#mscorlib/system/collections/objectmodel/readonlydictionary.cs,434

**Review by mjrousos — COMMENTED** [review]
This is probably fine, though just creating a new object also would work and it seems we've used that approach in the past.

What I'm more curious about is why we need this API implemented at all. How is it being used?

File: src/Microsoft.AspNetCore.SystemWebAdapters/Internal/NonGenericCollectionWrapper.cs (hunk 1 above)
Comment (twsouthwick) [reply src/Microsoft.AspNetCore.SystemWebAdapters/Internal/NonGenericCollectionWrapper.cs]: I've updated to use that pattern (although we don't need it to be thread-safe)

**Comment by twsouthwick** [comment]
> What I'm more curious about is why we need this API implemented at all. How is it being used?

Solely because `HttpContext.Items` is non generic and this is part of the API. If someone is using the API, it should be implemented since it's on the contract there.

**Review by mjrousos — APPROVED** [review]
LGTM

**Pull request merged**